code/script.js

```diff
@@ -231,95 +239,160 @@ const setSecret = () => {
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
+    }
+  } else if (category === 'accessories') {
+   
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Good guess! The person wears ${value}! Keep everyone with ${value}.`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `Sorry! The person doesn't wear ${value}! Remove everyone with ${value}.`
       )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
+  
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+    
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Good guess! The person has a ${value}! Keep everyone with ${value}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value));
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `Sorry! The person does not have a ${value}! Remove everyone with ${value}.`
+      )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value));
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
 
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard();
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  const result = window.confirm(`Are you sure?`);
+
+  if (result) {
+      checkMyGuess(personToConfirm);
+  }
+};
+
+
+// If the player wants to guess, invoke the checkMyGuess function.
+
 
 // If you confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
+  if (personToCheck === secret.name) {
+    alert(`Good job! Your guess on ${personToCheck} is correct! You win!`);
+    board.innerHTML = "";
```

Comment:
Since `board.innerHTML = ""` and` winOrLose.style.display = "flex"` are common to the if and else conditions, maybe it could be be written just once outside the if/else statements thus reducing your code lines.